nsq/async.py

```diff
@@ -241,7 +252,7 @@ def close(self):
     def _read_size(self, data):
         try:
             size = struct.unpack('>l', data)[0]
-            self.stream.read_bytes(size, self._read_body)
+            self._read_bytes(size, self._read_body)
         except Exception:
             self.close()
```

Comment:
You're right, good call.

I still think we should clarify that though, perhaps by moving the `_read_bytes` call out of the exception scope?